documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -834,6 +831,103 @@ public interface IJoystickInputHandler : IInputHandler
+
+This is an pseudocode example of what the Input Map system solves:
+
+Without inputMaps:
+```cs
+inputContext.Update();
+
+private Dictionary<string, bool> pressedKeys = new();
+
+inputContext.Gamepads.ThumbstickMove += @event => {Move(@event.Value)};
+inputContext.Joysticks.AxisMove += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+With inputMaps:
+```cs
+inputContext.Update();
+
+// This single line get rid of the two in the example above. TWO ISN'T THE LIMIT!
+inputContext.Map.GetAction("Move").Axis += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+## Implementation
+Each InputContext have one reference to an InputMap.
+This is how the InputMap class work:
+```cs
+public partial class InputMap
+{
+    public InputMapAction GetAction(string name);
+    public void AddAction(InputMapAction action, string name);
+    public void RemoveAction(string name);
+    public static InputMap Load(string path);
+    public static void Save(InputMap source, string path);
+    public static string Compile(InputMap source);
+}
+```
+
+`GetAction` method is the main entry point of the system. It allows the use to get an mapped action by it configurated name.
+
+```cs
+public partial class InputMapAction
+{
+    public bool IsPressed { get; }
+    public event Action<AxisInputEvent> Axis;
+}
+```
+
+## Configuration and saving
+The input map system's save system is one of the most important to it working. It helps to make the configuration easyer as faster.
+
+Instead of needing to hardcode all the actions, the user should create a file with a special syntax. This is an example of the sintax of the map system's saved data:
+```
+"move_up":
+    - KeyName.Up
+    - JoystickButton.DPadUp
+
+"move_down":
+- KeyName.Down
+- JoystickButton.DPadDown
+
+"move":
+    - Vector2D [
+        -KeyName.Down
+        -KeyName.Up
+        -KeyName.Left
+        -KeyName.Right
+    ]
+    - Joystick.Axis
+```
+
+The main syntax may somewhat resemble YAML, but it's much more symple and easy to parse.
+
+The main structure is based on a list, with it main feature having the line separation as the main way to separate the information. Also, the parsing is based on an finite-state model. The last data seted will recive the next configurations until a higher level information be setted.
+
+The label are wrote inside quotes ("") and ended with a collom (:). This starts a new `InputMapAction`.
+each new input are declared with a minnus sign (-) and a stace. After that, the name of the correct enum and the correct input.
+
+Input system can easly return vectors. after the minus sign, it's possible choose a vector type to recive (`Vector2D`, `Vector3D` or `Vector4D`) and the inputs that will math witch each axis, respectively.
```

Comment:
"return vectors" the example above uses keyboard buttons as well and does not specify in terms of requirements for how to map this into a vector. How is the axis matching done? (i.e. mathematically)

Reply:
Yeah, this parsing needs to be a lot more structured